parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -874,15 +890,39 @@ pub mod pallet {
 			);
 
 			<LatestSyncCommitteePeriod<T>>::set(period);
+			Self::prune_older_sync_committees();
 
 			Self::deposit_event(Event::SyncCommitteeUpdated { period });
 		}
 
+		// Contract: It is assumed that the light client do not skip sync committee.
+		fn prune_older_sync_committees() {
+			let threshold = T::SyncCommitteePruneThreshold::get();
+			let stored_sync_committees = <SyncCommittees<T>>::count();
+
+			if stored_sync_committees as u64 > threshold {
+				let latest_sync_committee_period = <LatestSyncCommitteePeriod<T>>::get();
+				let highest_period_to_remove = latest_sync_committee_period - threshold;
+
+				let mut current_sync_committee_to_remove = highest_period_to_remove;
+				let mut number_of_sync_committees_to_remove =
+					stored_sync_committees as u64 - threshold;
+
+				while number_of_sync_committees_to_remove > 0 {
+					<SyncCommittees<T>>::remove(current_sync_committee_to_remove);
+					number_of_sync_committees_to_remove =
+						number_of_sync_committees_to_remove.saturating_sub(1);
+					current_sync_committee_to_remove =
+						current_sync_committee_to_remove.saturating_sub(1);
+				}
+			}
```

Comment:
The problem here is the performance overhead of the while loop which could use up all the block weight.

Reply:
`while` loop would be most likely run only 1 time. As it runs after every insert.